Structure_Engine/Create/Elements/RetainingWall.cs

```diff
@@ -0,0 +1,127 @@
+        public static RetainingWall RetainingWall(Stem stem, PadFoundation footing, double retainedHeight, double coverDepth, double retentionAngle, double groundWaterDepth = 0)
+        {
+            if (stem.IsNull() || footing.IsNull())
+                return null;
+
+            return Query.IsValid(stem, footing) ? new RetainingWall()
+            {
+                Stem = stem,
+                Footing = footing,
+                RetainedHeight = retainedHeight,
+                CoverDepth = coverDepth,
+                RetentionAngle = retentionAngle,
+                GroundWaterDepth = groundWaterDepth
+            }
+            : null;
+        }
+
+        /***************************************************/
+
+        [Description("Create RetainingWall from a Line and defining properties.")]
+        [Input("line", "A Line parallell to the XY plane to build the RetainingWall from.")]
+        [Input("retainedHeight", "The retained height of soil measured from the bottom of the wall Footing.")]
+        [Input("normal", "Normal to the surface of the stem denoting the direction of the retained face.")]
+        [Input("stemThickness", "Thickness at the top and bottom of the stem.")]
+        [Input("toeLength", "Length of the toe of the footing.")]
+        [Input("heelLength", "Length of the heel of the footing.")]
+        [Input("footingThickness", "Thickness of the footing.")]
+        [Input("coverDepth", "The distance from top of Footing to finished floor level on the exposed face.")]
+        [Input("retentionAngle", "A property of the material being retained measured from the horizontal plane.")]
+        [Output("retainingWall", "RetainingWall with specified properties.")]
+        public static RetainingWall RetainingWall(Line line, double retainedHeight, Vector normal, double stemThickness, double toeLength,
+            double heelLength, ConstantThickness footingThickness, double coverDepth, double retentionAngle)
+        {
+            if (line.Start == null && line.End == null || line.IsNull())
+                return null;
+
+            if (!line.IsInPlane(Geometry.Create.Plane(line.Start, Vector.ZAxis)))
+            {
+                Base.Compute.RecordError("Provided line is not parallel to the XY plane. Please provide a line parallel to the XY plane.");
+                return null;
+            }
+
+            if (!line.IsInPlane(Geometry.Create.Plane(line.Start, Vector.ZAxis)))
+            {
+                Base.Compute.RecordError("Provided line is not parallel to the XY plane. Please provide a line parallel to the XY plane.");
+                return null;
+            }
+
+            PolyCurve stemOutline = new PolyCurve();
+            PolyCurve footingOutline = new PolyCurve();
+
+            //Create the footing outline. 
+            Line toeLine = line.ShallowClone();
+            Line heelLine = line.ShallowClone();
```

Comment:
Guess this does not really hurt, other than a minor, most likely unimportant slowdown, but do not need these clones. Translate returns a new line, and does not modify the incoming line.